runtime/src/plugin_builder.rs

```diff
@@ -150,12 +152,19 @@ impl<'a> PluginBuilder<'a> {
         self
     }
 
-    // Limit the number of instructions that can be executed
+    /// Limit the number of instructions that can be executed
     pub fn with_fuel_limit(mut self, fuel: u64) -> Self {
         self.fuel = Some(fuel);
         self
     }
 
+    /// Limit the size of the memory allocated up-front when
+    /// instantiating a module
+    pub fn with_static_memory_size(mut self, size: u64) -> Self {
```

Comment:
Considering the potential performance implications, and the big warnings from the underlaying wasmtime function's documentation, maybe it would be worth it to add a link to https://docs.wasmtime.dev/api/wasmtime/struct.Config.html#method.static_memory_maximum_size in this function's documentation